botocore/credentials.py

```diff
@@ -2180,6 +2182,15 @@ def load_credentials(self):
         # -js
         return None
 
+    def check_and_register_feature_id(self, method_name):
+        METHOD_FEATURE_MAP = {
```

Comment:
Nit: the capital letters here suggests that this is intended to be a constant, which would usually be defined on a class constant or module level.  

This winds up creating this dictionary every time, which isn't going to be super time intensive or anything, but if we are going to use this as a constant, I'd rather see this defined higher up on the class level so it's only defined once and reused every time we call this method.